### ISSUE TITLE
Add visual changelog to Switch Asset UI as to what will receive changes or not

### ISSUE THREAD

**Issue by BigRoy** [getavalon/core#389]
### Issue 

With #376 the Switch Asset UI no shows what the selected boxes "mean" for changes, but does not clearly showcase what it means for the end result on the selection. As such, it will be hard to figure out why some combinations may not be valid and result in no change - or what in the end the different containers will switch to.

### Proposal

It'd be best to include a list widget that shows what the current selection will mean for the final result, how will it be influenced.

Here's a *(very quickly!)* drawn mock-up showcasing how that could work:
![switch_ui_mockup](https://user-images.githubusercontent.com/2439881/58954755-56a6dc00-879a-11e9-8cbc-70648d7c60a4.jpg)

**Comment by BigRoy** [comment]
With #551 still being open for performance improvements I thought it would be just as good a time as any to see if we can optimize and get this issue done too. Here's a quick visual prototype.

![avalon_switch_asset_visual_view](https://user-images.githubusercontent.com/2439881/94699617-ce5b2b80-033a-11eb-9e69-7a2f1c23f1fe.gif)

Note that I've also taken the "preparation" code from #551 and improved the speed for that by making it simpler query with a single aggregation pipeline, thus a single query sent to the database - as I felt the time to when the popup showed still took some time with the new code. About another 30% speed improvement.

The changes made in #551 also mention a BUG introduced there reported by @tokejepsen where previously it allowed to choose a combination of changes that would NOT switch all selected assets but just a subset of them - which was not allowed anymore in #551. That original behavior and intention of the tool was also described [here](https://github.com/getavalon/core/issues/373#issuecomment-462312757) - specifically the switching to another shot, where the other shot could miss a specific subset and then still allow the change for all other pieces. I'm looking to revert the logic that causes the BUG so that it will indeed allow skipping some conversion but with the "change preview" it should also be clear as to what exactly is about to happen.

Pointers/ideas on how to easily improve how one can see what is about to change let me know. I'm not entirely happy with the two comparison lists.

Note to self: it could be worth looking into how Bulk/Batch Rename utilities provide previews of what is about to change. Or e.g. "refactoring" previews for code. Images are welcome!

@tokejepsen @davidlatwe @mkolar let me know if you have good references or ideas!

**Comment by tokejepsen** [comment]
Think this looks good!

Only comments are cosmetics from me:
- Orange instead of purple to keep with the general Avalon theme.
- The lists are big and only need a single line on each side.

**Comment by BigRoy** [comment]
> Orange instead of purple to keep with the general Avalon theme.

Yes, this was me being lazy and just copy pasting the first HTML color code that came from a google search, very lazy me.
I was thinking of even tinting each part of the string different, e.g. `asset` separate from `subset`, `representation`, etc.

> The lists are big and only need a single line on each side.

This was actually to accomodate for when switching multiple loaded content, like this:

![avalon_switch_asset_visual_view_multiple](https://user-images.githubusercontent.com/2439881/94820610-f0b77c80-0400-11eb-8fc9-51547d646423.gif)

Anyway, hope to find some time soon to continue with this :)

**Comment by tokejepsen** [comment]
Why is the representation red here?
![capture](https://user-images.githubusercontent.com/1860085/94821710-cdd59a00-03f9-11eb-940e-b124c37650e7.PNG)

**Comment by davidlatwe** [comment]
Would love to try out, but I am on vacation till next Monday.

Still, is this prototype can be `git checkout` somewhere in you fork ?
